src/main.nr

```diff
@@ -1,20 +1,30 @@
+use dep::std;
+use dep::u2b;
+
 global PRIME: u32 = 16777619;
 
 fn main(value: u32, hash_value: pub u32){
     let calculated_hash_value = calculate_hash(value);
-
     assert(calculated_hash_value == hash_value);
 }
 
 fn calculate_hash(num: u32) -> u32 {
```

Comment:
Since noir already does mod in case of overflow, you can also do something like this

```
fn calculate_hash(num: u32) -> u32 {

    let mut hash: u32 = 2166136261;     // 0x811C9DC5
    let bytes = u2b::u32_to_u8(num);

    for byte in bytes {
        let product = (hash * PRIME);
        hash = product ^ byte as u32;
    }
    hash as u32
}
```